app/models/govuk_publishing_components/component_doc.rb

```diff
@@ -0,0 +1,34 @@
+module GovukPublishingComponents
+  ComponentDoc = Struct.new(:id, :name, :description, :body, :fixtures) do
+    def self.get(id)
+      all.find { |component| component.id == id }
+    end
+
+    def self.all
+      fetch_component_doc.map { |component| build(component) }
+    end
+
+    def self.build(component)
+      fixtures = component[:fixtures].map { |id, data|
+        GovukPublishingComponents::ComponentFixture.new(id.to_s, data)
+      }
+
+      self.new(component[:id], component[:name], component[:description], component[:body], fixtures)
+    end
+
+    def fixture
+      fixtures.first
```

Comment:
This relies on convention, you’re right. I think this is ok for now. The convention of putting default first in the fixtures hasn't been an issue with static components.

The alternative would be to lookup via the `default` ID – if this was left out what should the guide page do? At the moment it'd fallback to the next fixture.

I think when there's a component generator this convention will be natural too.